src/hooks/useGeographicalStateFromRoute.ts

```diff
@@ -0,0 +1,16 @@
+import {useRoute} from '@react-navigation/native';
+import type {ParamListBase, RouteProp} from '@react-navigation/native';
+import {CONST as COMMON_CONST} from 'expensify-common/lib/CONST';
+
+type CustomParamList = ParamListBase & Record<string, Record<string, string>>;
+type State = keyof typeof COMMON_CONST.STATES;
+
+export default function useGeographicalStateFromRoute(stateParamName = 'state'): State | undefined {
```

Comment:
Could you add a docs/ comment to this hook to explain in plain english what it does, also ideally give an example